clang/lib/Driver/Driver.cpp

```diff
@@ -928,12 +928,18 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
   if (SYCLTargets && SYCLfpga)
     Diag(clang::diag::err_drv_option_conflict)
         << SYCLTargets->getSpelling() << SYCLfpga->getSpelling();
+
+  auto argSYCLIncompatible = [&](OptSpecifier OptId) {
+    if (!HasValidSYCLRuntime)
+      return;
+    if (Arg *IncompatArg = C.getInputArgs().getLastArg(OptId))
+      Diag(clang::diag::err_drv_fsycl_unsupported_with_opt)
+          << IncompatArg->getSpelling();
```

Comment:
Just as a note, an alternative to spawning the new diagnostic type could be to employ the same `err_drv_option_conflict` with "-fsycl" as an argument. However, the new option's text does appear more user-friendly, so no actual objections from my side.